### DISCUSSION TITLE
Using PostgREST from Android

### DISCUSSION THREAD

**Discussion by VinceT** [PostgREST/postgrest#3417] [Q&A] [answered]
Hi,
For purpose tests, I just made a small Web app using PHP. I access PostgREST thru curl commands using an SSL certificate + JWT.
I wrote the exact same app in pure HTML + JS + CSS. I can run "fetch" commands with GET, HEAD, PUT, PATCH, POST and DELETE methods. No need for a certificate  because I added my calling URL in the "server-cors-allowed-origins" parameter.
Then I tried to generate a mobile app by including this app within a Capacitor project. On my phone, the first page is displayed without error. So far, so good.
But... when I run any fetch request, I get this error message : 
8<------------------------------------------------------------
Failed to validate the certificate chain, error: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
8<------------------------------------------------------------

Question is : where / how do I add the certificate in order to make the generated Java run without error ?
I've read tons of things about CORS and also asked ChatGPT, but got no useful answer.

Thanks for your help.

**Comment by laurenceisla** [comment]
Hi... uhhh, I don't think this is a PostgREST issue, since it's built using Haskell and not Java. That means it may be a problem with your Android application. I found a similar issue here that could help you: https://github.com/axios/axios/issues/5271

Edit: From the link above, it could be because it expects an `https` source. Since PostgREST only serves `http` requests, then you may solve this by doing what's specified there or [using Nginx to add `https`](https://postgrest.org/en/v12/explanations/nginx.html#https).

**Comment by VinceT** [comment, marked as answer]
WOW !!!
What I did :
- revert PostgREST to http instead of https
- add http://localhost to server-cors-allowed-origins in postgres.conf
- restart PostgREST
- change "androidScheme": "https" to "androidScheme": "http" in MyPath\MyApp\capacitor.config.json
- add android:usesCleartextTraffic="true" to <application> tag in MyPath\MyApp\android\app\src\main\AndroidManifest.xml
And... everything works perfectly !
Thank you so much !
